### DISCUSSION TITLE
Im stuck with Lesson 7 challenge

### DISCUSSION THREAD

**Discussion by jistro** [Cyfrin/foundry-full-course-cu#277] [Q&A] [answered]
I tried to read the memory from the contract using the following command:
`cast storage  0xcf4fbA490197452Bd414E16D563623253eFb57D3 --rpc-url https://arbitrum-one.publicnode.com`
I received only this 
No matching artifacts found, fetching source code from Etherscan...

**Comment by jistro** [comment, marked as answer]
I found the solution it was in the code all the time

**Reply by 0xTagger** [reply]
Hey @jistro I been stuck for a while on this one. Any hints you can provide? Thanks.

**Reply by jistro** [reply]
> Hey @jistro I been stuck for a while on this one. Any hints you can provide? Thanks.

Far can I say, check one of the name of the variables, this variable has the number of the memory for the number, use the cast command I put and then transform into a uint

**Reply by 0xTagger** [reply]
thank you for the reply! I just figured it out 🙂

**Reply by mrMorningstar96** [reply]
Hi @jistro @0xTagger based on yours conversation I used following command:
```
cast storage 0xcf4fbA490197452Bd414E16D563623253eFb57D3  --rpc-url $SEPOLIA_RPC_URL --etherscan-api-key $ETHERSCAN_API_KEY
```
Then I used this:
```
cast index uint256 777 0xcf4fbA490197452Bd414E16D563623253eFb57D3
```
and got this as a result:
```
0x5943537dee02b87b05f294a89d86f7cd048f2df24f4d2b80cda539e8d54c15f8
```
I am not sure If I understood your tip correctly, any help would be appreciated.
Also If somehow I did correctly how do you transform result that I got into uint?

**Reply by AzmaeenGH** [reply]
@LukaNikolic96 , The address used in your code is not for Lesson 7 challenge of the foundry course.

**Comment by subhradeepsamantray** [comment]
Hello @jistro , can you please help me with how are you executing your commands for getting the answer? Like are you opening the contracts from etherscan into your local machine and then running the commands? I am not able to figure out how to approach to solve this challenge?
